Docsp 13857 (#16)
mongodb/docs-golang: source/usage-examples/insertOne.txt

=================

.. default-domain:: mongodb

> You can insert a single document in a collection by using the ``InsertOne()``

Reviewer: ```suggestion
You can insert a document in a collection by using the ``InsertOne()``
```
Nit: I think "a document" implies only one

method.

The following example creates a new document containing a ``title`` and ``text``

> field nad inserts it into the ``Haikus`` collection of the ``insertDB`` database:

Reviewer: ```suggestion
field and inserts it into the ``Haikus`` collection of the ``insertDB`` database:
```

.. note:: Non-existent Database and Collections

>     Databases and/or collections that don't exist will be created automatically
>     when you perform an insert operation against them.

Reviewer: ```suggestion
    The driver automatically creates databases and/or collections that don't exist
    when you perform an insert operation against them.
```
Suggestion: As is, it includes future tense and seems to be passive with "be created"

>     when you perform a write operation against them, if they don't exist.

Reviewer 2: ```suggestion
    when you perform a write operation against them if they don't already exist.
```

.. include:: /includes/usage-examples/run-example-tip.rst

.. literalinclude:: /includes/usage-examples/code-snippets/insertOne.go
   :start-after: begin insertOne
   :end-before: end insertOne
   :emphasize-lines: 4
   :language: go
   :dedent:

Click `here <{+example+}/insertOne.go>`__ to see a fully runnable example.

Expected Result
---------------

> After you run the code example, a new document with an ``_id`` that matches
> the output will be present:

Reviewer 2: Suggestion: Something about this wording is awkward. It's unclear where the matching document is "present"." I also don't think the other usage examples have really been referencing the output of the full code. Maybe:

```suggestion
After you run the code example, you should be able to find the following document in the ``haikus`` collection.
```

.. code-block:: json
   :copyable: false

   {
     _id: ObjectId("..."),

>      title: 'An Insert Haiku',
>      text: 'No bytes, no problem. Inserting a document. In MongoDB'

Reviewer 3: ```suggestion
     title: 'Record of a Shriveled Datum',
     text: 'No bytes, no problem. Inserting a document. In MongoDB',
     style:'haiku'
```

Nit: Can title be more haiku like? Got inspiration from [this wiki](https://en.wikipedia.org/wiki/Matsuo_Bash%C5%8D#List_of_works)

>      text: 'No bytes, no problem. Inserting a document. In MongoDB'

Reviewer 2: Suggestion: I think we should consistently use double quotes, as that's how documents appear in the Expected output section of the other usage examples. And consider putting the field names in quotes, I did that in my usage examples because its how it appears in Compass! See [UpdateOne](https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-13859-update-one-usage-ex/usage-examples/updateOne/#expected-result)
